tests/utests/types/string.c

```diff
@@ -0,0 +1,1327 @@
+    const struct lys_module *mod;
+    struct lysc_node_leaf *lysc_leaf;
+    struct lysp_node_leaf *lysp_leaf;
+    struct lysc_pattern *pattern;
+    struct lysc_range *range;
+
+    /* TEST BASE STRING */
+    schema = MODULE_CREATE_YANG("base", "leaf port {type string;}");
+    UTEST_ADD_MODULE(schema, LYS_IN_YANG, NULL, &mod);
+    assert_non_null(mod);
+    lysc_leaf = (void *) mod->compiled->data;
+    CHECK_LYSC_NODE_LEAF(lysc_leaf, NULL, 0, 0x5, 1, "port", 0, 0, 0, NULL, 0, 0, NULL, NULL);
+    CHECK_LYSC_TYPE_NUM((struct lysc_type_num *)lysc_leaf->type, LY_TYPE_STRING, 0, 0);
+    lysp_leaf = (void *) mod->parsed->data;
+    CHECK_LYSP_NODE_LEAF(lysp_leaf, NULL, 0, 0x0, 0, "port", 0, 0, NULL, 0, 0, NULL, NULL);
+    CHECK_LYSP_TYPE(&(lysp_leaf->type), 0, 0, 0, 0, 0, 0x0, 0, 0, "string", 0, 0, 1, 0, 0, 0);
+
+    /* TEST MODULE T0 */
+    schema = MODULE_CREATE_YANG("T0", "leaf port {type string"
+            "{length \"10 .. max\";}"
+            "}");
+    UTEST_ADD_MODULE(schema, LYS_IN_YANG, NULL, &mod);
+    assert_non_null(mod);
+    lysc_leaf = (void *) mod->compiled->data;
+    CHECK_LYSC_NODE_LEAF(lysc_leaf, NULL, 0, 0x5, 1, "port", 0, 0, 0, NULL, 0, 0, NULL, NULL);
+    CHECK_LYSC_TYPE_STR((struct lysc_type_str *)lysc_leaf->type, 0, 1, 0);
+    range = ((struct lysc_type_str *)lysc_leaf->type)->length;
+    CHECK_LYSC_RANGE(range, NULL, NULL, NULL, 0, 1, NULL);
+    assert_int_equal(range->parts[0].min_u64, 10);
+    assert_true(range->parts[0].max_u64 == 18446744073709551615ull);
+    lysp_leaf = (void *) mod->parsed->data;
+    CHECK_LYSP_NODE_LEAF(lysp_leaf, NULL, 0, 0x0, 0, "port", 0, 0, NULL, 0, 0, NULL, NULL);
+    CHECK_LYSP_TYPE(&(lysp_leaf->type), 0, 0, 0, 0, 0, 0x10, 0, 1, "string", 0, 0, 1, 0, 0, 0);
+    CHECK_LYSP_RESTR(lysp_leaf->type.length, "10 .. max", NULL, NULL, NULL, 0, NULL);
+
+    /* TEST MODULE T1 */
+    schema = MODULE_CREATE_YANG("T1", "leaf port {type string"
+            "{length \"min .. 20 | 50\";}"
+            "}");
+    UTEST_ADD_MODULE(schema, LYS_IN_YANG, NULL, &mod);
+    assert_non_null(mod);
+    lysc_leaf = (void *) mod->compiled->data;
+    CHECK_LYSC_NODE_LEAF(lysc_leaf, NULL, 0, 0x5, 1, "port", 0, 0, 0, NULL, 0, 0, NULL, NULL);
+    CHECK_LYSC_TYPE_STR((struct lysc_type_str *)lysc_leaf->type, 0, 1, 0);
+    range = ((struct lysc_type_str *)lysc_leaf->type)->length;
+    CHECK_LYSC_RANGE(range, NULL, NULL, NULL, 0, 2, NULL);
+    assert_int_equal(range->parts[0].min_u64, 0);
+    assert_int_equal(range->parts[0].max_u64, 20);
+    assert_int_equal(range->parts[1].min_u64, 50);
+    assert_int_equal(range->parts[1].max_u64, 50);
+    lysp_leaf = (void *) mod->parsed->data;
+    CHECK_LYSP_NODE_LEAF(lysp_leaf, NULL, 0, 0x0, 0, "port", 0, 0, NULL, 0, 0, NULL, NULL);
+    CHECK_LYSP_TYPE(&(lysp_leaf->type), 0, 0, 0, 0, 0, 0x10, 0, 1, "string", 0, 0, 1, 0, 0, 0);
+    CHECK_LYSP_RESTR(lysp_leaf->type.length, "min .. 20 | 50", NULL, NULL, NULL, 0, NULL);
+
+    // TEST MODULE T2
```

Comment:
use `/* */` instead of `//` comment style, it's not only here but in many places in the code